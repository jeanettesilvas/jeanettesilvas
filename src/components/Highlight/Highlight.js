import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch, faTimes } from '@fortawesome/free-solid-svg-icons';

import { getText } from '../../data';

import './Highlight.css';

export class Highlight extends Component {
    static propTypes = {
        onChange: PropTypes.func,
    };

    static defaultProps = {
        onChange: () => {},
    };

    state = {
        value: '',
    };

    text = getText();

    clear = () => {
        let value = '';
        this.setState({ value });
        this.props.onChange(value);
    };

    handleClearClick = event => {
        this.clear();
    };

    handleKeyDown = event => {
        if (event.which === 27) {
            this.clear();
        }
    };

    handleOnChange = event => {
        let value = event.target.value;
        this.setState({ value });
        this.props.onChange(value);
    };

    render() {
        let iconClass = 'highlight__icon';
        let searchIconClassName = classnames(
            iconClass,
            'highlight__icon--search',
            {
                hidden: !!this.state.value,
            }
        );

        let clearIconClassName = classnames(
            iconClass,
            'highlight__icon--clear',
            {
                hidden: !this.state.value,
            }
        );

        return (
            <div className="highlight">
                <input
                    className="highlight__input"
                    onChange={this.handleOnChange}
                    onKeyDown={this.handleKeyDown}
                    placeholder={this.text.highlight}
                    type="text"
                    value={this.state.value}
                />

                <FontAwesomeIcon
                    className={searchIconClassName}
                    icon={faSearch}
                />
                <FontAwesomeIcon
                    className={clearIconClassName}
                    icon={faTimes}
                    onClick={this.handleClearClick}
                />
            </div>
        );
    }
}
