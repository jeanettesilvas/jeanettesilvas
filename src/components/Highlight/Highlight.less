@import '../../less/_variables';

.highlight {
    display: relative;
    margin-top: 10px;

    &__input {
        background-color: @dark-gray;
        border: 1px solid @light-gray;
        border-radius: 3px;
        color: @light-gray-text;
        padding: @spacing-1x;
    }

    &__icon {
        color: @gray;
        position: absolute;
        right: 10px;
        top: 17px;
    }
}
