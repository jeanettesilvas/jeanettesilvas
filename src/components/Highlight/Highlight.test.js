import React from 'react';
import { shallow } from 'enzyme';
import { Highlight } from './Highlight';
import renderer from 'react-test-renderer';

describe('#render', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Highlight />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        const wrapper = shallow(<Highlight />);
        expect(wrapper.find('.highlight').length).toEqual(1);
    });

    it('hides the clear button when the input field is empty', () => {
        const wrapper = shallow(<Highlight />);

        expect(wrapper.find('.highlight__icon--clear.hidden').length).toEqual(
            1
        );

        expect(wrapper.find('.highlight__icon--search.hidden').length).toEqual(
            0
        );
    });

    it('hides the search icon when the input field is not empty', () => {
        const wrapper = shallow(<Highlight />);
        wrapper
            .find('.highlight__input')
            .simulate('change', { target: { value: 'asdf' } });

        expect(wrapper.find('.highlight__icon--clear.hidden').length).toEqual(
            0
        );

        expect(wrapper.find('.highlight__icon--search.hidden').length).toEqual(
            1
        );
    });
});
