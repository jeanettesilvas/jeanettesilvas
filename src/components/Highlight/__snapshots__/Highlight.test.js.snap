// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders correctly 1`] = `
<div
  className="highlight"
>
  <input
    className="highlight__input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="highlight tags"
    type="text"
    value=""
  />
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-search fa-w-16 highlight__icon highlight__icon--search"
    data-icon="search"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-times fa-w-11 highlight__icon highlight__icon--clear hidden"
    data-icon="times"
    data-prefix="fas"
    onClick={[Function]}
    role="img"
    style={Object {}}
    viewBox="0 0 352 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
